### PR TITLE
clean & new approach to getting files

### PR DESCRIPTION
**Pull request by n-r-w**
Hi! I'm going to include this package in the production code, but since there are certain requirements for imported packages, I will make appropriate changes to the code.

Another approach to getting files that seems more transparent than the existing one:
- Added FilesData structure
- The examples have been redesigned to use FilesData

General changes:
- Added a linter profile (golangci)
- Removed the dependency on the outdated errors package (github.com/pkg/errors => errors)
- Unnecessary exports have been removed
- Comments on public methods have been added
- "Magic numbers" replaced with constants
- Fixed gosec warnings (filepath.Clean)
- Added .gitignore, .gitattributes

### CODE REVIEW

**Comment by n-r-w** [comment]
This is breaking changes, but I don't think this is a problem, because the package is not very popular yet. Additionally, it makes sense to add a tag to fix the version.

File: file.go [hunk 1]
```diff
@@ -90,14 +92,14 @@ func ServeFile(server DownloadServer, contentType, path string) error {
...
-	return ServeContent(server, file, contentType, info.Name(), info.ModTime(), info.Size())
+	return serveContent(server, file, contentType, info.Name(), info.ModTime(), info.Size())
 }
 
-// ServeContent comes from http.ServeContent, and made some adaptations for DownloadServer
-func ServeContent(
-	server DownloadServer, content io.ReadSeeker, contentType, name string, modTime time.Time, size int64,
+// serveContent comes from http.serveContent, and made some adaptations for DownloadServer
+func serveContent( //nolint:gocognit
+	server downloadServer, content io.ReadSeeker, contentType, name string, modTime time.Time, size int64,
```
Comment (black-06) [inline file.go:102]: I recommend keeping `ServeContent` public, like `http.ServeContent`, so users can use io.ReadSeeker instead of just files

**Comment by black-06** [comment]
Wow, thank you very much 😘

**Comment by n-r-w** [comment]
> > This is breaking changes, but I don't think this is a problem, because the package is not very popular yet. Additionally, it makes sense to add a tag to fix the version.
> 
> Do you mean we add a tag to the current code, then merge it and add another tag?

Since the current master does not have a tag, there is no point in adding it. But it makes sense for new changes. 
Please write when you finish the review so that I correct all the comments at once.

File: upload.go [hunk 2]
```diff
@@ -31,52 +33,78 @@ func SaveMultipartFile(header *multipart.FileHeader, path string) error {
...
+	return &FilesData{form: form}, nil
+}
+
+// Get returns the files for the provided form key
+func (f *FilesData) Get(key string) []*multipart.FileHeader {
+	if headers := f.form.File[key]; len(headers) > 0 {
+		if len(headers) == 0 {
+			return nil
+		}
+
```
Comment (black-06) [inline upload.go:79]: It's never gonna get here.

File: upload.go [hunk 3]
```diff
@@ -31,52 +33,78 @@ func SaveMultipartFile(header *multipart.FileHeader, path string) error {
...
+		return headers
 	}
-	return nil, http.ErrMissingFile
+	return nil
+}
+
+// First returns the first file for the provided form key
+func (f *FilesData) First(key string) *multipart.FileHeader {
+	headers := f.Get(key)
+	if headers == nil {
```
Comment (black-06) [inline upload.go:88]:
```suggestion
	if len(headers) == 0 {
```

**Review by black-06 — APPROVED** [review]

File: upload.go [hunk 4]
```diff
@@ -31,52 +33,78 @@ func SaveMultipartFile(header *multipart.FileHeader, path string) error {
...
 
-// MultipartFormHeader returns the first file for the provided form key.
-// rpc request params must be 'stream google.api.HttpBody'.
-// It only applies to a single file, use ParseMultipartForm to process multiple files.
-func MultipartFormHeader(server UploadServer, key string) (*multipart.FileHeader, error) {
-	form, err := ParseMultipartForm(server)
+// FilesData is a wrapper around multipart.Form.
+type FilesData struct {
+	form *multipart.Form
+}
```
[the remark this answers is not in the document]
Comment (n-r-w) [reply upload.go:62]: Then I think it makes sense to change FilesData to FormData and add methods to access Value

[the remark this answers is not in the document]
Comment (black-06) [reply upload.go:62]: LGTM

**Comment by n-r-w** [comment]
Corrected all the comments in the new comit

**Pull request merged**